File: solution/3075. Maximize Happiness of Selected Children/3075.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& H, int k) {
        sort(H.rbegin(), H.rend());
        long res = 0;
        for (int i = 0; i < k && H[i] > i; ++i) {
            res += H[i] - i;
        }
        return res;
    }
};

File: solution/3075. Maximize Happiness of Selected Children/3075.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
        res = 0
        for i in range(k):
            if happiness[i] <= i:
                break
            res += happiness[i] - i
        return res
    
